File: backend/lib/backend/four_eyes.ex
defmodule Backend.FourEyes do
  @moduledoc """
  Runtime toggle for segregation-of-duties ("4-eyes") gates.

  Prod / test enforce the rule (approver ≠ preparer / qualifier /
  creator) across every gated flow: MO approve, CO director sign,
  PO director sign, vendor approve, customer approve.

  Dev flips ``config :backend, :enforce_four_eyes, false`` in
  ``config/dev.exs`` so a single developer can walk each lifecycle
  end-to-end from one seat without seeding a second user. The
  toggle exists ONLY to unblock local testing — never wire this
  key into ``runtime.exs`` or any release config.
  """

  @doc "True when the 4-eyes gate should be enforced (default)."
  def enforce?, do: Application.get_env(:backend, :enforce_four_eyes, true)
end

File: client/src/lib/four-eyes.ts
/**
 * Client mirror of the backend's `Backend.FourEyes` toggle.
 *
 * Set `NEXT_PUBLIC_ENFORCE_FOUR_EYES=false` in `.env.local` (dev only)
 * to let a single developer walk MO approve / CO director-sign /
 * customer approve flows end-to-end from one seat. Any other value
 * (including unset) enforces the 4-eyes rule — the correct default
 * for prod builds.
 *
 * Keep the dev config in sync with `psp/backend/config/dev.exs`
 * (`config :backend, :enforce_four_eyes, false`) — the server-side
 * gate is the real one; this flag only hides the pre-emptive UI
 * disables that would otherwise stop you from ever hitting the
 * button.
 */
export const ENFORCE_FOUR_EYES =
  process.env.NEXT_PUBLIC_ENFORCE_FOUR_EYES !== "false";
